lang/java/reef-common/src/main/java/org/apache/reef/runtime/common/REEFLauncher.java

```diff
@@ -0,0 +1,163 @@
+      final String[] args) throws BindException, IOException, InjectionException {
+
+    final JavaConfigurationBuilder commandLineBuilder =
+        Tang.Factory.getTang().newConfigurationBuilder();
+
+    new CommandLine(commandLineBuilder)
+        .registerShortNameOfClass(ClockConfigurationPath.class)
+        .processCommandLine(args);
+
+    return commandLineBuilder.build();
+  }
+
+  private static LaunchHelper getLaunchHelper(final String[] args) {
+    final Injector injector;
+    try {
+      injector = Tang.Factory.getTang().newInjector(processCommandLine(args));
+    } catch (final BindException | IOException | InjectionException e) {
+      throw fatal("Error in parsing the command line", e);
+    }
+
+    try {
+      return injector.getInstance(LaunchHelper.class);
+    } catch (InjectionException e) {
+      throw fatal("Unable to run LaunchClass", e);
+    }
+  }
+
+  private static RuntimeException fatal(final String msg, final Throwable t) {
+    LOG.log(Level.SEVERE, msg, t);
+    return new RuntimeException(msg, t);
+  }
+
+  private static RuntimeException fatal(REEFErrorHandler errorHandler, final String msg, final Throwable t) {
+    errorHandler.onNext(t);
+    LOG.log(Level.SEVERE, msg, t);
+    return new RuntimeException(msg, t);
+  }
+
+  private static Configuration readConfigurationFromDisk(LaunchHelper launchHelper) {
+    LOG.log(Level.FINEST, "Loading configuration file: {0}", launchHelper.getConfigurationPath());
+
+    final File evaluatorConfigFile = new File(launchHelper.getConfigurationPath());
+
+    if (!evaluatorConfigFile.exists()) {
+      final String message = "The configuration file " + launchHelper.getConfigurationPath() +
+          "doesn't exist. This points to an issue in the job submission.";
+      throw fatal(message, new FileNotFoundException());
+    } else if (!evaluatorConfigFile.canRead()) {
+      final String message = "The configuration file " + launchHelper.getConfigurationPath() +
+          " exists, but can't be read";
+      throw fatal(message, new IOException());
+    } else {
+      try {
+        return launchHelper.getConfigurationSerializer().fromFile(evaluatorConfigFile);
+      } catch (final IOException e) {
+        final String message = "Unable to parse the configuration file " +
+            launchHelper.getConfigurationPath();
+        throw fatal(message, e);
+      }
+    }
+  }
+
+  /**
+   * Launches a REEF client process (Driver or Evaluator).
+   *
+   * @param args
+   * @throws Exception
+   */
+  public static void main(final String[] args) {
+    LOG.log(Level.FINE, "Launcher started with user name [{0}]", System.getProperty("user.name"));
+
+    LOG.log(Level.FINE, "Launcher started. Assertions are {0} in this process.",
+        EnvironmentUtils.areAssertionsEnabled() ? "ENABLED" : "DISABLED");
+
+    final LaunchHelper launchHelper = getLaunchHelper(args);
+    launchHelper.getREEFVersion().logVersion();
```

Comment:
This is the only call to `getREEFVersion`. If we don't anticipate any others, we can just put a `logVersion` method into `LaunchHelper`